src/bench/logging.cpp

```diff
@@ -17,32 +25,67 @@ static void Logging(benchmark::Bench& bench, const std::vector<const char*>& ext
     bench.run([&] { log(); });
 }
 
-static void LoggingYoThreadNames(benchmark::Bench& bench)
+static void LogPrintLevelWithThreadNames(benchmark::Bench& bench)
 {
-    Logging(bench, {"-logthreadnames=1"}, [] { LogPrintf("%s\n", "test"); });
+    Logging(bench, {"-logthreadnames=1", "-debug=net"}, [] {
+        LogPrintLevel(BCLog::NET, BCLog::Level::Error, "%s\n", "test"); });
 }
-static void LoggingNoThreadNames(benchmark::Bench& bench)
+
+static void LogPrintLevelWithoutThreadNames(benchmark::Bench& bench)
 {
-    Logging(bench, {"-logthreadnames=0"}, [] { LogPrintf("%s\n", "test"); });
+    Logging(bench, {"-logthreadnames=0", "-debug=net"}, [] {
+        LogPrintLevel(BCLog::NET, BCLog::Level::Error, "%s\n", "test"); });
 }
-static void LoggingYoCategory(benchmark::Bench& bench)
+
+static void LogPrintWithCategory(benchmark::Bench& bench)
 {
     Logging(bench, {"-logthreadnames=0", "-debug=net"}, [] { LogPrint(BCLog::NET, "%s\n", "test"); });
 }
-static void LoggingNoCategory(benchmark::Bench& bench)
+
+static void LogPrintWithoutCategory(benchmark::Bench& bench)
 {
     Logging(bench, {"-logthreadnames=0", "-debug=0"}, [] { LogPrint(BCLog::NET, "%s\n", "test"); });
 }
-static void LoggingNoFile(benchmark::Bench& bench)
+
+static void LogPrintfCategoryWithThreadNames(benchmark::Bench& bench)
+{
+    Logging(bench, {"-logthreadnames=1", "-debug=net"}, [] {
+        LogPrintfCategory(BCLog::NET, "%s\n", "test");
+    });
+}
+
+static void LogPrintfCategoryWithoutThreadNames(benchmark::Bench& bench)
+{
+    Logging(bench, {"-logthreadnames=0", "-debug=net"}, [] {
+        LogPrintfCategory(BCLog::NET, "%s\n", "test");
+    });
+}
+
+static void LogPrintfWithThreadNames(benchmark::Bench& bench)
```

Comment:
In 102b2033493f0d61e9763d094cb8a0017f7e3a10 bench: order the logging benchmark code by output: 

Given that the code-movement, and re-ordering the `BENCHMARK(` calls doesn't change the output, this seems like a no-op?

In general, is (re-)ordering code in a `.cpp` file based on the order functions are called something we do?

Reply:
Benchmarks are registered into a `std::map` so they are alphabetically sorted anyways https://github.com/bitcoin/bitcoin/blob/master/src/bench/bench.h#L68